system/security/securesocket/securesocket.cpp

```diff
@@ -326,7 +327,31 @@ class CSecureSocket : implements ISecureSocket, public CInterface
         int pending = SSL_pending(m_ssl);
         if(pending > 0)
             return pending;
-        return m_socket->avail_read();
+        // pending == 0 : check if there still might be data to read
+        // (often used as a check for if socket was closed by peer)
+        size32_t avr = m_socket->avail_read();
+        if (avr > 0)
+        {
+            // bytes may be SSL/TLS protocol and not part of msg
+            byte c[2];
+            // TODO this may block ...
+            pending = SSL_peek(m_ssl, c, 1);
+            // 0 almost always means socket was closed
+            if (pending == 0)
+                return 0;
+            if (pending > 0)
+                return SSL_pending(m_ssl);
+            // pending < 0 : TODO should handle SSL_ERROR_WANT_READ/WRITE error
+            if (m_loglevel >= SSLogNormal)
+            {
+                int ret = SSL_get_error(m_ssl, pending);
```

Comment:
Am I missing something?  You don't seem to be using anything from this call.  I don't think ERR_get_error() requires SSL_get_error() to have been called?

Reply:
es, thanks.  I meant to log the ret error.  I'll fix.

Reply:
added ret to log.